15.md

```diff
@@ -4,18 +4,14 @@ NIP-15
 End of Stored Events Notice
 ---------------------------
 
-`final` `optional` `author:Semisol`
+`final` `mandatory` `author:Semisol`
 
-Relays may support notifying clients when all stored events have been sent.
+Relays MUST support notifying clients when all stored events have been sent.
 
-If a relay supports this NIP, the relay SHOULD send the client a `EOSE` message in the format `["EOSE", <subscription_id>]` after it has sent all the events it has persisted and it indicates all the events that come after this message are newly published.
+The relay MUST send the client an `EOSE` message in the format `["EOSE", <subscription_id>]` after it has sent all the events it has persisted and it indicates all the events that come after this message are newly published.
 
 Client Behavior
 ---------------
 
-Clients SHOULD use the `supported_nips` field to learn if a relay supports end of stored events notices.
-
-Motivation
-----------
-
-The motivation for this proposal is to reduce uncertainty when all events have been sent by a relay to make client code possibly less complex.
+Clients may use the `supported_nips` field to learn if a relay supports end of stored events notices.
+Clients MUST unsubscribe when they get an `EOSE` message from the relay.
```

Comment:
Actually it's better to just delete this entirely.

Reply:
You mean all the client behavior section?
no need to unsubscribe and close connection on client side?

Reply:
I've added your suggestions. but I think it's better to unsubscribe on EOSE, and create a new subscription later just if we want to refetch new events based on our client behavior. some clients might want to refetch only on hard page refresh. some might want to refetch every n seconds.

Reply:
> I've added your suggestions. but I think it's better to unsubscribe on EOSE, and create a new subscription later just if we want to refetch new events based on our client behavior. some clients might want to refetch only on hard page refresh. some might want to refetch every n seconds.

nostr events are transmitted in real time. Sounds like you want to force the standard HTTP request pattern over it, but nostr being based on WebSockets real-time is a feature not a bug.